internal/pkg/override/cdk.go

```diff
@@ -140,11 +139,20 @@ func (cdk *CDK) cleanUp(body []byte) ([]byte, error) {
 }
 
 // ScaffoldWithCDK bootstraps a CDK application under dir/ to override the seed CloudFormation resources.
+// If the directory is not empty, then returns an error.
 func ScaffoldWithCDK(fs afero.Fs, dir string, seeds []template.CFNResource) error {
+	// Ignore errors on purpose. IsEmpty returns false if the directory does not exist, hence we should only check
+	// if the dir is empty if it also exists.
+	exists, _ := afero.Exists(fs, dir)
+	isEmpty, _ := afero.IsEmpty(fs, dir)
+	if exists && !isEmpty {
+		return fmt.Errorf("directory %q is not empty", dir)
+	}
```

Comment:
```suggestion
	if isEmpty, _ := afero.IsEmpty(fs, dir); !isEmpty {
		return fmt.Errorf("directory %q is not empty", dir)
	}
```
does `exists` even matter?

Reply:
actually maybe i'm misunderstanding because of the error. should there be a different error when the directory doesn't exist? like `directory doesn't exist`? or should that not even be an error case?

Reply:
oh nvm. i understand the comment now lol. good work 👍

Reply:
- If the directory does not exist -> 👌 it's all good . We will create the directory for them and write files.
- If the directory exists and is empty -> 👌 it's all good. We will write the files to the directory.
- If the directory exists but is not empty -> 💥bad. We don't want to overwrite any files that they have in the directory.

Sadly, the following does not work:
```
isEmpty, _ := afero.IsEmpty(fs, dir); !isEmpty
```
That's what I had initially, but you can see the implementation here: https://github.com/spf13/afero/blob/d92c300a1b6eaa1690b7aecab2ef7fd0d64a94e1/util.go#L285
if the directory does not exist, then `afero.IsEmpty` returns false.

Reply:
I've updated the comment slightly, hopefully that makes it a bit clearer :) I was surprised that `IsEmpty` returned false when the path did not exist, but I can also see why afero maintainers implement it this way.